Dockerfile

```diff
@@ -19,7 +19,7 @@ RUN apk add --no-cache \
 
 # Download go dependencies
 WORKDIR /osmosis
-COPY go.mod go.sum ./
```

Comment:
cc @niccoloraspa @p0mvn is there some more correct way I should be doing this?

AFAIU, we may actually be building against the wrong osmomath rn, since we can cache the osmomath copy. 

We need it here though, since go mod download looks for a local osmomath replace

Reply:
Commented below with suggestions, rest LGTM on a first pass